service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDataService.java

```diff
@@ -44,24 +49,13 @@
     static final int USERLAYER_BASE_LAYER_ID = PropertyUtil.getOptional(USERLAYER_BASELAYER_ID, -1);
     static final int USERLAYER_MAX_FEATURES_COUNT = PropertyUtil.getOptional(USERLAYER_MAXFEATURES_COUNT, -1);
 
-    /**
-     * @param gjsWorker geoJSON and featurecollection items
-     * @param user      oskari user
-     * @param fparams   user given attributes for layer
-     * @return user layer data in user_layer table
-     */
-
-    public UserLayer storeUserData(GeoJsonWorker gjsWorker, User user, Map<String, String> fparams) throws ServiceException {
-
-
-        final UserLayer userLayer = new UserLayer();
-        final UserLayerStyle style = new UserLayerStyle();
-        List <UserLayerData> userLayerDataList = new ArrayList<UserLayerData>();
-
-        log.info("user data store start: ", fparams);
-
-        //TODO: Style insert
+    public UserLayer storeUserData(SimpleFeatureCollection fc, User user, Map<String, String> fparams) throws ServiceException {
```

Comment:
I would rather replace fparams with UserLayer and parse the form fields on the ActionHandler level.

Reply:
Or create a bunch of parseUserLayer(), parseStyle() methods that you use to get a class out of params and call userLayerService.insertUserLayer(userLayer, style, userLayerDataList) outside of those params to classes methods.